src/internal/agent/hooks/pods_test.go

```diff
@@ -154,3 +154,27 @@ func TestPodMutationWebhook(t *testing.T) {
 		})
 	}
 }
+func TestGetImageAnnotationKey(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		containerName string
+		expectedKey   string
+	}{
+		{
+			containerName: "nginx",
+			expectedKey:   "zarf.dev/original-image-nginx",
+		},
+		{
+			containerName: "a-very-long-container-name-that-exceeds-sixty-three-characters",
+			expectedKey:   "zarf.dev/original-image-a-very-long-container-name-that-exceeds-sixty-th",
```

Comment:
note that this expected key is 72 characters long

Reply:
we should be expecting 63 character maximum, right?

Reply:
Yeah that is expected. Read this https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/#syntax-and-character-set. Annotations keys are made of two segments. The name segment must be 63 characters or less. The prefix must be not longer than 253 characters.

Reply:
Ah! Interesting that they separate the two.